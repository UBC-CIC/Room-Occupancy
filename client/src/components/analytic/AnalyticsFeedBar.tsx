import * as React from "react";
import { Segment, Header, Dimmer, Loader, Progress } from "semantic-ui-react";
import { useGetOccupancy } from "../../shared/hooks/useGetOccupancy";
import { AnalyticsFeed } from "./AnalyticsFeed";
import { toFriendlyTime } from "../../shared/helpers/utility";

export function AnalyticsFeedBar(props: any) {
  const { occupancyList } = useGetOccupancy();
  return (
    <Segment>
      <Header as="h4">
        <Header.Content>
          Occupancy Feed
          <Header.Subheader>Updates every 10 seconds</Header.Subheader>
        </Header.Content>
      </Header>
      <Segment basic style={{ overflow: "auto", maxHeight: "75vh" }}>
        {occupancyList.length === 0 ? (
          <Progress percent={100} indicating>
            Loading
          </Progress>
        ) : (
          occupancyList.map((data: any, key: any) => {
            return (
              <AnalyticsFeed
                key={key}
                summaryHeader={
                  "Camera " + data?.Data[0].ScalarValue + " Occupancy:"
                }
                summarySubheader={data?.Data[4].ScalarValue}
                summaryDate={""}
                icon={data?.Data[1].ScalarValue}
                metaHeader={toFriendlyTime(data?.Data[3].ScalarValue)}
              />
            );
          })
        )}
      </Segment>
    </Segment>
  );
}
