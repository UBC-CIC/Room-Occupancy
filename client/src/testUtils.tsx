import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import {
  MemoryRouter,
  MemoryRouterProps,
  BrowserRouter,
} from "react-router-dom";

// Implement with creation of providers
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <MemoryRouter>{children}</MemoryRouter>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// Usage example:
// test("matches snapshot", () => {
//   // Replace 'Navbar' with your desired component and provide the props
//   testSnapshot(Navbar, { /* props for Navbar */ });
// });
const testComponentSnapshot = <P extends {}>(
  Component: React.ComponentType<P>,
  props: P,
  routerProps: MemoryRouterProps = {}
) => {
  const { asFragment } = render(
    <MemoryRouter {...routerProps}>
      <Component {...props} />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
};

export * from "@testing-library/react";
export { customRender, testComponentSnapshot };
