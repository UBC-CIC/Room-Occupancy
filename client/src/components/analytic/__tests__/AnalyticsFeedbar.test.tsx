import React from "react";
import "@testing-library/jest-dom";
import { customRender } from "../../../testUtils";
import { AnalyticsFeedBar } from "../AnalyticsFeedBar";

describe("<AnalyticsFeedBar />", () => {
  test("should matchSnapshot", () => {
    const { asFragment } = customRender(
      <AnalyticsFeedBar>Colin</AnalyticsFeedBar>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
