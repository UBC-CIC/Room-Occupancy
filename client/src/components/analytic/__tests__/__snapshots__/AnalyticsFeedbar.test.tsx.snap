// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalyticsFeedBar /> should matchSnapshot 1`] = `
<DocumentFragment>
  <div
    class="ui segment"
  >
    <h4
      class="ui header"
    >
      <div
        class="content"
      >
        Occupancy Feed
        <div
          class="sub header"
        >
          Real-time updates
        </div>
      </div>
    </h4>
    <div
      class="ui basic segment"
      style="overflow: auto; max-height: 75vh;"
    >
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               5 people
              <div
                class="date"
              >
                5 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 2 Occupancy 
              </a>
               10 people
              <div
                class="date"
              >
                2 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               6 people
              <div
                class="date"
              >
                5 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               7 people
              <div
                class="date"
              >
                1 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               5 people
              <div
                class="date"
              >
                5 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 2 Occupancy 
              </a>
               10 people
              <div
                class="date"
              >
                2 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               6 people
              <div
                class="date"
              >
                5 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               7 people
              <div
                class="date"
              >
                1 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               5 people
              <div
                class="date"
              >
                5 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 2 Occupancy 
              </a>
               10 people
              <div
                class="date"
              >
                2 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               6 people
              <div
                class="date"
              >
                5 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               7 people
              <div
                class="date"
              >
                1 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               5 people
              <div
                class="date"
              >
                5 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 2 Occupancy 
              </a>
               10 people
              <div
                class="date"
              >
                2 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               6 people
              <div
                class="date"
              >
                5 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui feed"
      >
        <div
          class="event"
        >
          <div
            class="content"
          >
            <div
              class="summary"
            >
              <a
                class="user"
              >
                Zone 1 Occupancy 
              </a>
               7 people
              <div
                class="date"
              >
                1 minutes ago
              </div>
            </div>
            <div
              class="meta"
            >
              <a
                class="like"
              >
                <i
                  aria-hidden="true"
                  class="icon"
                />
                Camera ID: A5S7GB-UBC
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
